test/unit/test_dmap.py

```diff
@@ -0,0 +1,294 @@
+
+
+class TestDmapRead(unittest.TestCase):
+    """
+    Testing class for DmapRead class
+    """
+    def setUp(self):
+        pass
+
+    """
+    Testing DmapRead's constructor
+    """
+    def test_incorrect_path(self):
+        """
+        Testing DmapRead's constructor with an non-existant folder.
+
+        Expected behaviour: raise FileNotFoundError
+        """
+        self.assertRaises(FileNotFoundError, pydarn.DmapRead,
+                          './dog/somefile.rawacf')
+
+    def test_incorrect_file(self):
+        """
+        Tests if DmapRead's constructor with an non-existant file
+
+        Expected bahaviour: raises FileNotFoundError
+        """
+        self.assertRaises(FileNotFoundError, pydarn.DmapRead,
+                          '../testfiles/somefile.rawacf')
+
+    def test_empty_file(self):
+        """
+        Tests if DmapRead's constructor with an empty file
+
+        Expected behaviour: raise EmptyFileError
+        """
+        self.assertRaises(pydarn.dmap_exceptions.EmptyFileError,
+                          pydarn.DmapRead, '../testfiles/empty.rawacf')
+
+    def test_open_dmap_file(self):
+        """
+        Tests DmapRead's constructor on opening a rawacf.
+        It should be able to open the file, read it and convert to bytearray.
+
+        Checks:
+            - bytearray instance is created from reading in the file
+            - bytearray is not empty
+        """
+        file_path = fitacf_file
+        dm = pydarn.DmapRead(file_path)
+        self.assertIsInstance(dm.dmap_bytearr, bytearray)
+        self.assertGreater(dm.dmap_end_bytes, 0)
+
+    def test_integrity_check_dmap_file(self):
+        """
+        Tests DmapRead test_initial_data_integrity
+        It should be able to read through the bytearray quickly
+        ensureing no curruption has occured in the file.
+
+        Behaviours: raising no exceptions
+        """
+        file_path = rawacf_file
+        dm = pydarn.DmapRead(file_path)
+        dm.test_initial_data_integrity()
+
+    def test_read_dmap_file(self):
+        """
+        Tests DmapRead test read_dmap.
+
+        Behaviour: raising no exceptions
+        """
+        file_path = fitacf_file
+        dm = pydarn.DmapRead(file_path)
+        dmap_records = dm.read_records()
+        self.assertIsInstance(dmap_records, collections.deque)
+        self.assertIsInstance(dmap_records[0], collections.OrderedDict)
+        self.assertIsInstance(dmap_records[4]['bmnum'], pydarn.DmapScalar)
+        self.assertIsInstance(dmap_records[1]['ptab'], pydarn.DmapArray)
+        self.assertIsInstance(dmap_records[7]['channel'].value, int)
+        self.assertIsInstance(dmap_records[2]['ltab'].value, np.ndarray)
+        self.assertEqual(dmap_records[0]['ptab'].dimension, 1)
+        self.assertEqual(dmap_records[50]['gflg'].value[1], 0)
+
+    # TODO: Again dependent on the file used :/
+    def test_integrity_check_corrupt_file1(self):
```

Comment:
I think it would be a good idea to just force a corruption of a file here. I think its possible to just read raw binary and then scramble a couple bytes

Reply:
I do that essentially in `def test_dmap_read_corrupt_stream(self):` similar to how you had it in your backscatter code. This was more can I detect an actual **blacklisted** file that is corrupt.

Reply:
I do that essentially in def test_dmap_read_corrupt_stream(self): similar to how you had it in your backscatter code. This was more can I detect an actual blacklisted file that is corrupt.

Reply:
Sounds good. I would suggest a more apt name might be something like "test_read_blacklist_file"